src/pages/register.tsx

```diff
@@ -0,0 +1,208 @@
+                                        <a href="https://www.gov.uk/help/privacy-notice" className="govuk-link">
+                                            Privacy
+                                        </a>
+                                        &nbsp;and&nbsp;
+                                        <a href="https://www.gov.uk/help/cookies" className="govuk-link">
+                                            Cookies
+                                        </a>
+                                        &nbsp;policies
+                                    </p>
+                                    <div className="govuk-checkboxes">
+                                        <div className="govuk-checkboxes__item">
+                                            <input
+                                                className="govuk-checkboxes__input"
+                                                id="checkbox-user-research"
+                                                name="checkbox-user-research"
+                                                type="checkbox"
+                                                value="checkUserResearch"
+                                            />
+                                            <label
+                                                className="govuk-label govuk-checkboxes__label"
+                                                htmlFor="checkbox-user-research"
+                                            >
+                                                If you are willing to be contacted as part of user research, please tick
+                                                this box.
+                                            </label>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                            <input
+                                type="submit"
+                                name="createAccount"
+                                value="Create account"
+                                id="create-account-button"
+                                className="govuk-button"
+                            />
+                            <input value={regKey} type="hidden" name="regKey" />
+                        </form>
+                    </div>
+                    <div className="govuk-grid-column-one-thirds">
+                        <h3 className="govuk-heading-m">Already have an account?</h3>
+                        <a href="/signin" className="govuk-link">
+                            Sign in
+                        </a>
+                    </div>
+                </div>
+            </main>
+        </Layout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContext): {} => {
+    const cookies = parseCookies(ctx);
+    const userCookie = cookies[USER_COOKIE];
+
+    const errors: ErrorInfo[] = [];
+
+    const { key } = ctx.query;
```

Comment:
In the case where key doesn't exist we may want to redirect to the requestAccess page